src/dlr_relayvm.cc

```diff
@@ -240,16 +248,25 @@ void RelayVMModel::GetOutput(int index, void* output) {
 }
 
 void RelayVMModel::GetOutputShape(int index, int64_t* shape) const {
-  CHECK_LT(index, num_outputs_) << "Output index is out of range.";
-  std::memcpy(shape, outputs_[index]->shape, sizeof(int64_t) * outputs_[index]->ndim);
+  CHECK_LT(index, output_shapes_.size()) << "Output index is out of range.";
+  if (index < outputs_.size()) {
```

Comment:
What are these two cases for? When would index be >= outputs.size()?

Reply:
For dynamic models the output shapes are dynamic and depends on inference.  In the pre inference case the output shapes are obtained from metadata and after inference the output shapes are obtained from outputs.

Reply:
I see, so calling inference will initialize `outputs_`, otherwise it will use the metadata shapes from `output_shapes_`? To make it clearer, what do you think about doing something like:
```
if (outputs.empty()) {
  // Inference has not been called yet. Get shapes from metadata.
  CHECK_LT(index, output_shapes_.size()) << "Output index is out of range.";
  std::copy(output_shapes_[index].begin(), output_shapes_[index].end(), shape);
} else {
  CHECK_LT(index, outputs_.size()) << "Output index is out of range.";
  std::memcpy(shape, outputs_[index]->shape, sizeof(int64_t) * outputs_[index]->ndim);
}
```